src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2423,6 +2423,24 @@ namespace winrt::TerminalApp::implementation
     // - Paste text from the Windows Clipboard to the focused terminal
     void TerminalPage::_PasteText()
     {
+        // First, check if we're in broadcast input mode. If so, let's tell all
+        // the controls to paste.
+        if (const auto& tab{ _GetFocusedTabImpl() })
+        {
+            if (tab->TabStatus().IsInputBroadcastActive())
+            {
+                tab->GetRootPane()->WalkTree([](auto&& pane) {
+                    if (auto control = pane->GetTerminalControl())
+                    {
+                        control.PasteTextFromClipboard();
```

Comment:
Idle thought... should this not go via the same path as RawWriteString? PasteTextFromClipboard means "ask the app for the clipboard content," which could result in one ContentDialog per control (!) confirming the paste.

I'd expect this:

```mermaid
sequenceDiagram
    TerminalPage-->>TermControl: PasteTextFromClipboard
    TermControl-->>TerminalPage: GetTextFromClipboard
    TerminalPage-->>TermControl: Here's the text bud
    TermControl-->>Pane: SentString("here's the...")
    Pane-->>OtherTermControl: RawWriteString("here's the...")
    Pane-->>OtherTermControl: RawWriteString("here's the...")
    Pane-->>OtherTermControl: RawWriteString("here's the...")
```

Reply:
note from discussion: bracketed paste, make sure that works

Reply:
Hmm. This is... less trivial. 

`TerminalPage::_PasteFromClipboardHandler` calls the event's `HandleClipboardData`, which is a callback that `Interactivity` set up to call `_core->PasteText` [^1] when the paste is completed. However, Core & Interactivity don't actually know anything about broadcast mode. That's all good and well.

I suppose this could be done by finagling the paste handling a bit. We'd bubble up to TermControl, it'd register the paste callback itself, and in its own handler it could decide to re-raise a `event Windows.Foundation.TypedEventHandler<Object, StringSentEventArgs> PasteSent;`[^2], which pane could listen to, and then propagate. Feels kinda extra dirty, tbh.

I don't think the dialog will pop multiple times, on account of the fact that there can only ever be one dialog. I'll have to validate that though. 

Alternatively, we could do the broadcast check inside `TerminalPage::_PasteFromClipboardHandler` and just call `WalkTree([text](){ p->Control().PasteText(); }); and plumb that through. ~Might be tricky to determine that though. `TermControl` doesn't expose the `Interactivity` out of it, so we couldn't compare the `sender`~ well, I suppose if we're not in broadcast mode, we could just call the event callback as before. 

Thoughts?


[^1]: `core->PasteText` is what handles bracketed paste, so that's fine.
[^2]: can't be the StringSent, cause we do need to special-case pastes

Reply:
@DHowett gibe thoughts pls, so I can merge this when I get back from break for 1.19 selfhost ☺️

Reply:
We chatted last-ish week about RawWriteString taking a "StringKind" of Clipboard, Raw, Whatever. Noting that here to close the loop. :)